### ISSUE TITLE
Skip Validation is not working with json style of storage-config secret.

### ISSUE THREAD

**Issue by Jooho** [kserve/kserve#3263]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

**What did you expect to happen:**
When a user uses json style of storage-config, skip verification option does not work.
which was working with  `serving.kserve.io/s3-verifyssl`  annotation.

json style
~~~
apiVersion: v1
stringData:
  localMinIO: |
    {
      "type": "s3",
      "access_key_id": "admin",
      "secret_access_key": "password",
      "endpoint_url": "http://minio.minio.svc:9000",
      "bucket": "modelmesh-example-models",
      "region": "us-south"
      "verify": "0"   <===
    }
kind: Secret
metadata:
  name: storage-config
type: Opaque    
~~~

annotation style
~~~
apiVersion: v1
data:
  AWS_ACCESS_KEY_ID: VEhFQUNDRVNTS0VZ
  AWS_SECRET_ACCESS_KEY: VEhFUEFTU1dPUkQ=
kind: Secret
metadata:
  annotations:
    serving.kserve.io/s3-endpoint: minio.minio.svc:9000
    serving.kserve.io/s3-region: us-east-2
    serving.kserve.io/s3-useanoncredential: "false"
    serving.kserve.io/s3-usehttps: "1"
    serving.kserve.io/s3-verifyssl: "0"        <===
  name: storage-config
  namespace: kserve-demo
type: Opaque
~~~

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version:
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by israel-hdez** [comment]
For StorageSpec the JSON is copied to equivalent environment variables. That's done here: https://github.com/kserve/kserve/blob/6f4bf79e4dd8bd15fb920508b2652bb20222ad9f/python/kserve/kserve/storage/storage.py#L114-L120. There is no copying done for any `verify` field.

The only place where I can find the structure of the storage secret is here: https://github.com/kserve/kserve/tree/master/docs/samples/storage/storageSpec#create-common-storage-secret. There is no mention of a `verify` field, so I'm not sure if this can be seen as a bug, yet I agree that it is desirable to have a field to turn off SSL verification.

**Comment by sivanantha321** [comment]
fixed by #3267
